Mathlib/Algebra/Lie/Derivation/Basic.lean

```diff
@@ -290,4 +292,27 @@ instance instLieAlgebra : LieAlgebra R (LieDerivation R L L) where
 
 end
 
+section
+
+variable {R L : Type*} [Field R] [LieRing L] [LieAlgebra R L]
+
+/-- The linear embedding of Lie derivations into linear endormophisms. -/
+def toLinearMapLinearMap : (LieDerivation R L L) →ₗ[R] (L →ₗ[R] L) where
+  toFun := toLinearMap
+  map_add' := by intro D1 D2; dsimp
+  map_smul' := by intro D1 D2; dsimp
+
+/-- The linear embedding of Lie derivations into linear endormophisms is injective. -/
+lemma toLinearMapLinearMap_injective : Function.Injective ((@toLinearMapLinearMap R L).toFun) := by
+  dsimp [toLinearMapLinearMap]
+  intro D1 D2 h
+  ext a
+  exact congrFun (congrArg DFunLike.coe h) a
+
+/-- Lie derivations over a finite dimensional Lie algebra form a finite dimensional module. -/
+instance instModuleFinite [Module.Finite R L] : Module.Finite R (LieDerivation R L L) :=
+  FiniteDimensional.of_injective toLinearMapLinearMap toLinearMapLinearMap_injective
```

Comment:
```suggestion
instance instNoetherian [IsNoetherian R L] : IsNoetherian R (LieDerivation R L L) :=
  isNoetherian_of_linearEquiv (LinearEquiv.ofInjective _ (toLinearMapLieHom_injective R L)).symm
```
For emphasis, given the above you'll have:
```lean
example {K : Type*} [Field K] [LieAlgebra K L] [Module.Finite K L] :
    Module.Finite K (LieDerivation K L L) :=
  inferInstance
```

Reply:
Done in https://github.com/leanprover-community/mathlib4/pull/11927/commits/9d4f6c6b8d96bd7e75e44dfa5c64caad5fbc520f

Reply:
Btw feel free to hit the "Resolve conversation" button if you are confident that you have addressed a reviewer's suggestion.